hnn_core/network.py

```diff
@@ -966,12 +968,16 @@ def add_tonic_bias(self, *, cell_type=None, amplitude=None,
             'tstop': tstop
         }
 
-    def _add_cell_type(self, cell_name, pos):
+    def _add_cell_type(self, cell_name, pos, cell_template=None):
         """Add cell type by updating pos_dict and gid_ranges."""
         ll = self._n_gids
         self._n_gids = ll + len(pos)
         self.gid_ranges[cell_name] = range(ll, self._n_gids)
+
         self.pos_dict[cell_name] = pos
+        if cell_template is not None:
```

Comment:
In what situations would an empty `cell_template` be passed?

Reply:
This is for the input drives ... so we use the same function for both drives and cells. I think fundamentally drives and cells are not too different and we should try to unify them as much as possible

Reply:
Ah that makes sense. Same goes for the drive connections, I think the easiest situation would be getting rid of `_NetworkDrive` and just using `_Connectivity` internally. Perhaps with a `is_drive=bool` attribute.

Reply:
Maybe just replace `_NetworkDrive['conn']` to start with ? I wonder if you can replace `_create_drive_conns` etc. with `add_connection`